File: 준영/LV2/타겟 넘버.py
result = 0

def dfs(n, numbers, d, target):
    global result
    if n == len(numbers):
        if d == target:
            result += 1
    else:
        for j in [-1, 1]:
            dfs(n+1, numbers, d+(numbers[n]*j), target)


def solution(numbers, target):
    global result 

    dfs(0, numbers, 0, target)

    answer = result
    return answer

File: 준영/LV2/타겟 넘버.js
var result = 0

function dfs(n, numbers, d, target) {
  if (n === numbers.length) {
    if (d === target) {
      result += 1
    }
  } else {
    for (const i of [-1, 1]) {
      dfs(n+1, numbers, d+(numbers[n]*i), target)
    }
  }
}

function solution(numbers, target) {
  dfs(0, numbers, 0, target)
  var answer = result
  return answer
}